lib/dynflow/dispatcher.rb

```diff
@@ -0,0 +1,58 @@
+module Dynflow
+  module Dispatcher
+    Request = Algebrick.type do
+      Event = type do
+        fields! execution_plan_id: String,
+                step_id:           Fixnum,
+                event:             Object
+      end
+
+      Execution = type do
+        fields! execution_plan_id: String
+      end
+
+      Ping = type do
+        fields! receiver_id: String
+      end
+
+      variants Event, Execution, Ping
+    end
+
+    Response = Algebrick.type do
+      variants Accepted = atom,
+               Failed   = type { fields! error: String },
+               Done     = atom,
+               Pong     = atom
+    end
+
+    Envelope = Algebrick.type do
+      fields! request_id: Integer,
+              sender_id: String,
+              receiver_id: type { variants String, AnyExecutor = atom, UnknownWorld = atom },
+              message: type { variants Request, Response }
```

Comment:
This is something I've not originally planned for, I like it though. How do you get the type of `receiver_id` field. How does it `to_s`?